src/opensynth/models/faraday/gaussian_mixture/model.py

```diff
@@ -245,8 +191,6 @@ def sample(self, num_datapoints: int) -> torch.Tensor:
         return torch.cat(result, dim=0)
 
     def _get_component_precision(self, component: int) -> torch.Tensor:
-        if self.covariance_type == "tied":
-            return self.precisions_cholesky
         return self.precisions_cholesky[component]
 
 
```

Comment:
K-means model is redundant (since we are now using the sk-learn k-means) so you can delete the code below this line in `model.py`. Also might want to rename this script to `gmm_model.py` to not confuse it with the `model.py` script in the parent directory?